src/content/guides/tree-shaking.md

```diff
@@ -176,6 +176,141 @@ T> Note that any imported file is subject to tree shaking. This means if you use
 
 Finally, `"sideEffects"` can also be set from the [`module.rules` configuration option](/configuration/module/#module-rules).
 
+## Clarifying tree shaking and `sideEffects`
+
+The [`sideEffects`](/configuration/optimization/#optimizationsideeffects) and [`usedExports`](/configuration/optimization/#optimizationusedexports) (more known as tree shaking) optimizations are two different things.
+
+__`sideEffects` is much more effective__ since it allows to skip whole modules/files and the complete subtree.
+
+`usedExports` relies on [terser](https://github.com/terser-js/terser) to detect side effects in statements. It is a difficult task in JavaScript and not as effective as straighforward `sideEffects` flag. It also can't skip subtree/dependencies since the spec says that side effects need to be evaluated. While exporting function works fine, react's Higher Order Components (HOC) are problematic in this regard.
```

Comment:
```suggestion
`usedExports` relies on [terser](https://github.com/terser-js/terser) to detect side effects in statements. It is a difficult task in JavaScript and not as effective as straighforward `sideEffects` flag. It also can't skip subtree/dependencies since the spec says that side effects need to be evaluated. While exporting function works fine, React's Higher Order Components (HOC) are problematic in this regard.
```